restcomm/restcomm.dao/src/main/java/org/mobicents/servlet/restcomm/entities/Geolocation.java

```diff
@@ -0,0 +1,676 @@
+        return dateExecuted;
+    }
+
+    public Sid getAccountSid() {
+        return accountSid;
+    }
+
+    public String getSource() {
+        return source;
+    }
+
+    public String getDeviceIdentifier() {
+        return deviceIdentifier;
+    }
+
+    public GeolocationType getGeolocationType() {
+        return geolocationType;
+    }
+
+    public String getResponseStatus() {
+        return responseStatus;
+    }
+
+    public String getCellId() {
+        return cellId;
+    }
+
+    public String getLocationAreaCode() {
+        return locationAreaCode;
+    }
+
+    public Integer getMobileCountryCode() {
+        return mobileCountryCode;
+    }
+
+    public String getMobileNetworkCode() {
+        return mobileNetworkCode;
+    }
+
+    public Long getNetworkEntityAddress() {
+        return networkEntityAddress;
+    }
+
+    public Integer getAgeOfLocationInfo() {
+        return ageOfLocationInfo;
+    }
+
+    public String getDeviceLatitude() {
+        return deviceLatitude;
+    }
+
+    public String getDeviceLongitude() {
+        return deviceLongitude;
+    }
+
+    public Long getAccuracy() {
+        return accuracy;
+    }
+
+    public String getPhysicalAddress() {
+        return physicalAddress;
+    }
+
+    public String getInternetAddress() {
+        return internetAddress;
+    }
+
+    public String getFormattedAddress() {
+        return formattedAddress;
+    }
+
+    public DateTime getLocationTimestamp() {
+        return locationTimestamp;
+    }
+
+    public String getEventGeofenceLatitude() {
+        return eventGeofenceLatitude;
+    }
+
+    public String getEventGeofenceLongitude() {
+        return eventGeofenceLongitude;
+    }
+
+    public Long getRadius() {
+        return radius;
+    }
+
+    public String getGeolocationPositioningType() {
+        return geolocationPositioningType;
+    }
+
+    public String getLastGeolocationResponse() {
+        return lastGeolocationResponse;
+    }
+
+    public String getCause() {
+        return cause;
+    }
+
+    public String getApiVersion() {
+        return apiVersion;
+    }
+
+    public URI getUri() {
+        return uri;
+    }
+
+    public enum GeolocationType {
+        Immediate("Immediate"), Notification("Notification");
+
+        private final String glt;
+
+        private GeolocationType(final String glt) {
+            this.glt = glt;
+        }
+
+        public static GeolocationType getValueOf(final String glt) {
+            GeolocationType[] values = values();
+            for (final GeolocationType value : values) {
+                if (value.toString().equals(glt)) {
+                    return value;
+                }
+            }
+            throw new IllegalArgumentException(glt + " is not a valid application GeolocationType.");
+        }
+
+        @Override
+        public String toString() {
+            return glt;
+        }
+    };
+
+    public Geolocation setSid(Sid sid) {
```

Comment:
Well DTO is a design pattern usually used in conjunction with DAO pattern. I just assumed you were using it, but apparently your Entities are more VOs than DTOs. In that case, having final classes is mandatory. Funny thing is equals/hashCode is not implemented properly for all of them so they are actual VOs. Apparently there is a mix of DTOs and VOs there, anyway i can see this is the established Design again ;)